github/GithubObject.py

```diff
@@ -45,23 +62,29 @@ def __repr__(self):
 
 NotSet = _NotSetType()
 
+T = TypeVar("T")
+
+Opt = Union[T, _NotSetType]
+
 
-class _ValuedAttribute:
-    def __init__(self, value):
+class _ValuedAttribute(Generic[T]):
+    value: T
+
+    def __init__(self, value: T):
         self.value = value
 
 
 class _BadAttribute:
```

Comment:
this can be a subclass of `_ValuedAttribute`, so we don't need to type `Union[_BadAttribute, _ValuedAttribute]`

Reply:
your
 suggestion looks good， I'll implement it